src/runtime/internal/ssr.ts

```diff
@@ -69,20 +69,34 @@ export function merge_ssr_styles(style_attribute, style_directive) {
 	return style_object;
 }
 
-export const escaped = {
-	'"': '&quot;',
-	"'": '&#39;',
-	'&': '&amp;',
-	'<': '&lt;',
-	'>': '&gt;'
-};
+const ATTR_REGEX = /[&"]/g;
+const CONTENT_REGEX = /[&<]/g;
+
+/**
+ * Note: this method is performance sensitive and has been optimized
+ * https://github.com/sveltejs/svelte/pull/5701
+ */
+export function escape(html: string, is_attr = false) {
+	if (typeof html !== 'string') return html;
+
+	const pattern = is_attr ? ATTR_REGEX : CONTENT_REGEX;
+	pattern.lastIndex = 0;
+
+	let escaped = '';
+	let last = 0;
+
+	while (pattern.test(html)) {
+		const i = pattern.lastIndex - 1;
+		const ch = html[i];
+		escaped += html.substring(last, i) + (ch === '&' ? '&amp;' : (ch === '"' ? '&quot;' : '&lt;'));
+		last = i + 1;
+	}
 
-export function escape(html) {
-	return String(html).replace(/["'&<>]/g, match => escaped[match]);
+	return escaped + html.substring(last);
 }
 
 export function escape_attribute_value(value) {
-	return typeof value === 'string' ? escape(value) : value;
+	return typeof value === 'string' ? escape(value, true) : value;
```

Comment:
This isn't needed anymore since it's handled by `escape`:

```suggestion
	return escape(value, true);
```

Reply:
if I accept your suggestion above then `escape` would no longer be doing this and this line should stay though, right?  so I think we'd need to figure out which of the two changes is the better one to make

Reply:
I guess it depends on how it's used elsewhere. The previous implementation of `escape_attribute_value` includes the same attack vector if it's ever used on untrusted objects.